core/src/test/java/hudson/UtilTest.java

```diff
@@ -781,4 +782,20 @@ public void testPermissionsToMode() throws Exception {
         assertEquals(0000, Util.permissionsToMode(PosixFilePermissions.fromString("---------")));
     }
 
+    @Test
+    public void testDifferenceDays() {
+        Date day1_10am = new Date(2018, 4, 6, 10, 0);
+        Date day1_11pm55 = new Date(2018, 4, 6, 23, 55);
+        Date day2_01am = new Date(2018, 4, 7, 1, 0);
+        Date day2_11pm = new Date(2018, 4, 7, 1, 0);
+        Date day3_08am = new Date(2018, 4, 8, 8, 0);
+        assertEquals(0, Util.differenceInCalendarDay(day1_10am, day1_11pm55));
```

Comment:
It might be simpler to use a parameterized test for this, but it'd have to be in a separate class thanks to junit 4.